### ISSUE TITLE
under Android environment, the select(...) function canot support operation on stdin-like file descriptors

### ISSUE THREAD

**Issue by mxi1** [lkl/linux#387]
I find that, under Android enviornment the `select(...)` function cann't support operation on stdin-like file descriptors, and will return the following error message:
```
the following error occurred for selecting: Operation not supported on transport endpoint
```
This issue causes some apps (e.g. telnet, nc, and other user-interactive apps) not work.

**Comment by thehajime** [comment]
I guess the above error is from the following code.

https://github.com/lkl/linux/blob/master/tools/lkl/lib/hijack/hijack.c#L295

```
	/* FIXME: handle mixed case of hostfd and lklfd */
	if (lklfds && hostfds)
		return lkl_set_errno(-LKL_EOPNOTSUPP);
```

This is due to descriptors monitored by caller are from host's fd (less than LKL_FD_OFFSET) and LKL's fd (more than LKL_FD_OFFSET), which is the current limitation of LKL.  Other multiplexers (poll/epoll) also have the same limitation.

If this is the case, it's not android-related issue but a generic one.

I have work-in-progress patch for epoll but not ready for various cases.

**Comment by mxi1** [comment]
@thehajime  Thanks your reply. 

Should the standard input/ouput/error output( 0,1, and 2 ) be treated specially? The lkl-hijacked apps might need to access these 3 file descriptors.

**Comment by thehajime** [comment]
> Should the standard input/ouput/error output( 0,1, and 2 ) be treated specially? The lkl-hijacked apps might need to access these 3 file descriptors.

I agree but this requires the same logic that all other host descriptors will be select(2)-ed properly.

What I have in my mind and did in my WIP branch is do both host_select() and lkl_sys_select() first, and aggregate the result into the original select(2) to the caller.  In this way, hijack library would be transparent to the application which uses host descriptors (sockets, files, etc).

And this is how rump hijack library (NetBSD) handles this situations.

https://github.com/rumpkernel/src-netbsd/blob/buildrump-src/lib/librumphijack/hijack.c#L2006
